### PR TITLE
Improve CI configuration

### PR DESCRIPTION
**Pull request by sergiught**
### 🔧 Changes

We're bringing several improvements to the CI configuration within this PR:

- Added govulncheck step
- Updated golangci-lint config
- Refactored Makefile

### 📝 Checklist

- [x] All new/changed/fixed functionality is covered by tests (or N/A)
- [x] I have added documentation for all new/changed functionality (or N/A)

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -19,25 +19,17 @@ jobs:
...
-          files=$(git status --porcelain)
-          if [[ -n ${files}  ]]; then
-            >&2 echo "Go generate resulted in changed files:"
-            >&2 echo ""
-            >&2 echo "${files}"
-            >&2 echo ""
-            >&2 echo "Please run \`go generate ./...\` to regenerate code."
-            >&2 git diff
-            exit 1
-          fi
```
Comment (sergiught) [inline .github/workflows/main.yml:34]: This is now under the `make check-docs` command.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -19,25 +19,17 @@ jobs:
...
+        run: make check-docs
+
+      - name: Check for go vulnerabilities
+        run: make check-vuln
 
       - name: Check for linting errors
         uses: golangci/golangci-lint-action@v3
         with:
-          version: v1.50.1
-          args: -c .golangci.yml --allow-parallel-runners
```
Comment (sergiught) [inline .github/workflows/main.yml:40]: We're pushing the `--allow-parallel-runners` flag inside the `.golangci.yml` file.

File: main.go [hunk 3]
```diff
@@ -19,8 +19,6 @@ func main() {
 	schema.DescriptionKind = schema.StringMarkdown
 
 	plugin.Serve(&plugin.ServeOpts{
-		ProviderFunc: func() *schema.Provider {
-			return provider.New()
-		},
+		ProviderFunc: provider.New,
```
Comment (sergiught) [inline main.go:22]: Unnecessary use of lambda, replaced with direct call to `provider.New`

File: internal/recorder/http_recorder.go [hunk 4]
```diff
@@ -98,13 +98,13 @@ func redactHeaders(i *cassette.Interaction) {
 }
 
 func redactDomain(i *cassette.Interaction, domain string) {
-	i.Request.Host = strings.Replace(i.Request.Host, domain, RecordingsDomain, -1)
-	i.Request.URL = strings.Replace(i.Request.URL, domain, RecordingsDomain, -1)
+	i.Request.Host = strings.ReplaceAll(i.Request.Host, domain, RecordingsDomain)
```
Comment (sergiught) [inline internal/recorder/http_recorder.go:101]: Fixing gocritic mentions.

File: scripts/golangci-lint.sh [hunk 5]
```diff
@@ -1,10 +0,0 @@
-#!/usr/bin/env bash
-
-set -eu -o pipefail
```
Comment (sergiught) [inline scripts/golangci-lint.sh:3]: No need for this file any more, this was pushed inside the Makefile.

**Review by Widcket — APPROVED** [review]

**Pull request merged**